include/aikido/util/yaml_utils.hpp

```diff
@@ -0,0 +1,202 @@
+            % MatrixType::RowsAtCompileTime % rows));
+    }
+
+    if (node.Tag() == "Vector" || node.Tag() == "!Vector") {
+        matrix.resize(rows, 1);
+        for (Index i = 0; i < rows; ++i) {
+            matrix(i, 0) = node[i].template as< _Scalar >();
+        }
+    } else if (node.Tag() == "Matrix" || node.Tag() == "!Matrix") {
+        Index const cols = node[0].size();
+
+        if (MatrixType::ColsAtCompileTime != Eigen::Dynamic 
+         && cols != MatrixType::ColsAtCompileTime) {
+            throw runtime_error(str(
+                format("Matrix has incorrect number of cols: expected %d; got %d.")
+                % MatrixType::ColsAtCompileTime % cols));
+        }
+
+        matrix.resize(rows, cols);
+
+        for (Index r = 0; r < node.size(); ++r) {
+            if (node[r].Type() != YAML::NodeType::Sequence) {
+                throw runtime_error(str(
+                    format("Row %d of the matrix must be a sequence.") % r));
+            } else if (node[r].size() != cols) {
+                throw runtime_error(boost::str(
+                    format("Expected row %d to have %d columns; got %d.")
+                        % r % cols % node[r].size()));
+            }
+
+            for (Index c = 0; c < cols; ++c) {
+                matrix(r, c) = node[r][c].template as<_Scalar>();
+            }
+        }
+    } else {
+        throw runtime_error(str(
+            format("Unknown type of matrix '%s'.") % node.Tag()));
+    }
+}
+
+template <class _Scalar, int Dim, int Mode, int _Options>
+inline void deserialize(YAML::Node const &node,
+                        Eigen::Transform<_Scalar, Dim, Mode, _Options> &pose)
+{
+    deserialize(node, pose.matrix());
+}
+
+template <class T>
+inline bool has_child(YAML::Node const &node, T const &key)
+{
+	return node[key].IsDefined();
+}
+
+} // namespace util
+} // namespace aikido
+
+
+namespace YAML {
+namespace detail {
+
+template <typename MatrixType, bool IsVectorAtCompileTime>
+struct encode_impl {
+  static Node encode(MatrixType const &matrix)
+  {
+    assert(false && "Unknown MatrixType.");
+  }
+};
+
+template <typename MatrixType>
+struct encode_impl<MatrixType, true> {
+  static Node encode(MatrixType const &matrix)
+  {
+    typedef typename MatrixType::Index Index;
+
+    Node node;
+    node.SetTag("Vector");
+
+    for (Index i = 0; i < matrix.size(); ++i) {
+      node.push_back(Node(matrix[i]));
+    }
+    return node;
+  }
+};
+
+template <typename MatrixType>
+struct encode_impl<MatrixType, false> {
+  static Node encode(MatrixType const &matrix)
+  {
+    typedef typename MatrixType::Index Index;
+
+    Node node;
+    node.SetTag("Matrix");
+
+    for (Index r = 0; r < matrix.rows(); ++r)
+    {
+      Node row(NodeType::Sequence);
+
+      for (Index c = 0; c < matrix.cols(); ++c)
+      {
+          row.push_back(matrix(r, c));
+      }
+
+      node.push_back(row);
+    }
+    return node;
+  }
+};
+
+} //detail
```

Comment:
Nit: We _may_ want to move this helper code into the `detail` directory. @gilwoolee @jslee02 Thoughts?